WordPress/Classes/Services/MediaCoordinator.swift

```diff
@@ -194,6 +200,21 @@ class MediaCoordinator: NSObject {
         return mediaProgressCoordinator.media(withIdentifier: uploadID)
     }
 
+    /// Returns an existing media objcect with the specificed objectID
+    ///
+    /// - Parameter objectID: the object unique ID
+    /// - Returns: an media object if it exists.
+    ///
+    func media(withObjectID objectID: String) -> Media? {
+        guard let storeCoordinator = mainContext.persistentStoreCoordinator,
+            let url = URL(string: objectID),
+            let managedObjectID = storeCoordinator.managedObjectID(forURIRepresentation: url),
+            let media = mainContext.object(with: managedObjectID) as? Media else {
```

Comment:
I think we want `existingObject(with:)` instead of `object(with:`) – `existingObject` returns nil if the object doesn't exist, whereas `objectWith` creates a fault and never returns nil.